pytorch_lightning/trainer/trainer.py

```diff
@@ -633,14 +635,15 @@ def run_train(self) -> None:
             if not self.interrupted:
                 self.state = TrainerState.INTERRUPTED
                 self.on_keyboard_interrupt()
-        except (RuntimeError, AssertionError):
-            # if an exception is raised, the finally block is executed and can hide the actual exception
-            # that was initially raised if `on_train_end` also raises an exception. we want to avoid that
-            # for assertions and other runtime errors so we aren't misled while debugging
+                self.accelerator.on_train_end()
```

Comment:
Why do we use `accelerator.on_train_end` instead of `train_loop` here and below?

Reply:
train_loop.on_train_end would call accelerator.on_train_end internall also, but I suspect it is removed because they want to avoid checkpointing on this interruption?

Reply:
Yes, the reason is avoid the "troublesome" checkpointing (when there is error for some ranks but not all, hence hang in broadcast).

Reply:
or the case where we fail during training, but the checkpoint callback is configured for a monitor available only in validation